runelite-client/src/main/java/net/runelite/client/plugins/praydose/PrayerDosePlugin.java

```diff
@@ -112,16 +115,15 @@ public Overlay getOverlay()
 	public void onTick(GameTick tick)
 	{
 		overlay.onTick();
+	}
 
+	@Subscribe
+	public void onItemContainerChanged(final ItemContainerChanged event)
+	{
 		this.holdingPrayerPotion = false;
 		this.holdingSuperRestore = false;
 		this.holyWrench = false;
 
-		if (client.getGameState() != GameState.LOGGED_IN)
-		{
-			return;
-		}
-
 		this.checkInventory();
```

Comment:
Are the events atomic or are they like grouped by tick and subject to race conditions? Slightly concerned there might be some de-sync if you equipped a ring of the gods, for example, and the equipment container event ran before the inventory container event. The first would pick up the holy wrench in the equipment, but the second would not pick up anything in your inventory, and that would throw off the prayer calculations. I think I can just split the holy wrench flag into two pieces to indicate whether it's present in your inventory or equipment, but I'm not sure if that's necessary.

Reply:
Grouped by tick

Reply:
You should at least check if this is either inventory or equipment and then continue, because otherwise you will get million of events when opening bank

Reply:
Got it. Changed it to only process equipment and inventory events. Also went ahead and split up the two holy wrench statuses / container checks, so it's a bit less processing on each event. Unfortunately don't have an equipable holy wrench item to test with, but I can't forsee any issues with it.